src/unistd.rs

```diff
@@ -2504,7 +2506,7 @@ impl User {
                 }
             } else if Errno::last() == Errno::ERANGE {
                 // Trigger the internal buffer resizing logic.
-                reserve_buffer_size(&mut cbuf, cbuf_max)?;
+                reserve_double_buffer_size(&mut cbuf, bufsize)?;
```

Comment:
The sets the initial and maximum sizes to the same value.  You should probably choose a larger maximum size.  As I said, FreeBSD internally uses 1MB.

Reply:
And how we can inquiry the OS to get this information? Blindly put a number does not sounds right.